librosa/util/utils.py

```diff
@@ -627,8 +636,26 @@ def normalize(S, norm=np.inf, axis=0):
     else:
         raise ParameterError('Unsupported norm: {}'.format(repr(norm)))
 
+    # get the vectors-ids smaller than the threshold
+    small_values_idx = (length < threshold)
+
     # Avoid div-by-zero
-    length[length < SMALL_FLOAT] = 1.0
+    length[small_values_idx] = 1.0
+
+    if mode and np.sum(small_values_idx):
+        # init empty slice object
+        slices = [slice(None)] * S.ndim
+
+        # set slice for target-axis
+        target_axis_idx = int(np.logical_not(axis))
+        slices[target_axis_idx] = np.squeeze(small_values_idx)
+
+        if mode == 'fill_zeros':
+            # zero-out vectors
+            S[slices] = 0
+        if mode == 'fill_constant':
+            # fill with unit vectors
+            S[slices] = normalize(np.ones([S.shape[axis], 1]), norm=norm, axis=axis)[0]
```

Comment:
This is a destructive operation.  If you're going to modify S in-place, do so on a copy.